state_machine/applications/system-check/tests/imu_test.py

```diff
@@ -0,0 +1,262 @@
+"""
+Python system check script for PyCubed satellite board
+PyCubed Mini mainboard-v02 for Pocketqube Mission
+IMU Sensor Test
+* Author(s): Yashika Batra
+"""
+
+import time
+from ulab import numpy
+from lib import pycubed as cubesat
+
+
+wait_time = 2
+norm = numpy.linalg.norm
+
+
+def stationary_user_test():
+    """
+    All user interaction for the stationary test happens in this function
+    Set wait times, prompt the users and gather IMU data and input
+    """
+    # print user prompts
+    print("Please leave the cubesat stationary on a table.")
+    start_test = input("Type Y to start the stationary test, any key" +
+                       " to cancel: ")
+
+    # if user opts to cancel the test, return None, handle in the next function
+    if start_test.lower() != "y":
+        return None, None, None
+    print("Collecting IMU data...")
+
+    # collect acceleration and gyroscope readings, return None mag default
+    time.sleep(wait_time)
+    acc = cubesat.acceleration()
+    gyro = cubesat.gyro()
+    print("Data Collection Complete")
+    return acc, gyro, None
+
+
+def stationary_imu_test(result_dict):
+    """
+    All stationary test automation happens in this function
+    Process user test results and update the result dictionary accordingly
+    """
+    # record acceleration
+    acc, gyro, mag = stationary_user_test()
+    if acc is None and gyro is None:
+        # user entered "n", cancel the test
+        result_dict['IMU_AccStationary'] = (
+            "Stationary test not completed.", False)
+        result_dict['IMU_GyroStationary'] = (
+            "Stationary test not completed.", False)
+        return result_dict
+
+    # else continue running the test
+    result_val_string_acc = f"Acc: {acc}"
+    result_val_string_gyro = f"Gyro: {gyro}"
+
+    # if total acceleration ~= 9.8 m/s^2 and gyro ~= 0 deg/s, true
+    result_val_bool_acc = 9.8 - norm(acc) < 0.2
+    result_val_bool_gyro = norm(gyro) < 0.2
+    result_dict['IMU_AccStationary'] = (
+        result_val_string_acc, result_val_bool_acc)
+    result_dict['IMU_GyroStationary'] = (
+        result_val_string_gyro, result_val_bool_gyro)
+    return result_dict
+
+
+def moving_user_test():
```

Comment:
perhaps we could combine these `*_user_test()` into a `request_imu_data(prompt)` that prints the prompt, then on `y` reads the imu after half a second of waiting. I feel like that would reduce the code duplication.

Reply:
Perhaps it would be better to just return the IMU readings in an array after reading from the imu for `n` seconds with something like 10 reads per second.

Reply:
Fixed! I'm still setting the wait time as 5 seconds, because a good amount of these tests involve user involvement, and I think 1 second is too little for the user to do things like rotate/move the cubesat or move a magnet.

Reply:
changed it to 3 seconds - I tested out the script and it felt like this was enough time for the user to react to prompts (and not so long that it got annoying)

Reply:
Sounds great!